src/core/rating_store.py

```
@@ -0,0 +1,203 @@
+            )
+            for field, value in [
+                ("channel_informative", channel_informative),
+                ("channel_balanced", channel_balanced),
+                ("channel_sourced", channel_sourced),
+                ("channel_entertaining", channel_entertaining),
+            ]:
+                # value != 0 → setzen, value == 0 → auf NULL zurücksetzen
+                # (damit Abwahl einer vorherigen Bewertung wirkt)
+                db_value = value if value != 0 else None
+                conn.execute(
+                    f"UPDATE analyses SET {field} = ? WHERE id = ?",
```

Ruff (0.14.14)

[error] Possible SQL injection vector through string-based query construction

(S608)